tests/test_classes.py

```diff
@@ -0,0 +1,119 @@
+        with server.register_modules():
+            self.assertTrue(execute_agent(**kwargs)[1])
+            self.assertTrue(execute_agent(**kwargs_classic)[1])
+        self.assertFalse(execute_agent(**kwargs)[1])
+        self.assertFalse(execute_agent(**kwargs_classic)[1])
+
+    def test_sc_module(self):
+        class TestAgent(ScAgent):
+            def on_event(self, init_element: ScAddr, init_edge: ScAddr, action_element: ScAddr) -> ScResult:
+                finish_action_with_status(action_element, True)
+                return ScResult.OK
+
+        def is_executing_successful(i: int) -> bool:
+            return execute_agent(
+                arguments={},
+                concepts=[],
+                initiation=f"agent{i}",
+                wait_time=WAIT_TIME,
+            )[1]
+
+        agent1 = TestAgent("agent1", ScEventType.ADD_OUTGOING_EDGE)
+        agent2 = TestAgent("agent2", ScEventType.ADD_OUTGOING_EDGE)
+        agent3 = TestAgent("agent3", ScEventType.ADD_OUTGOING_EDGE)
+
+        module1 = ScModule(agent1)
+        module2 = ScModule(agent2)
+        server.add_modules(module1)
+        module1.add_agent(agent3)
+        with server.register_modules():
+            module1.add_agent(agent2)  # Cannot add agent to module in register time
+            self.assertTrue(is_executing_successful(1))
+            self.assertFalse(is_executing_successful(2))
+            self.assertTrue(is_executing_successful(3))
+
+            server.add_modules(module2)
+            self.assertTrue(is_executing_successful(2))
+
+            server.remove_modules(module1)
+            module1.remove_agent(agent3)
+            server.add_modules(module1)
+            self.assertTrue(is_executing_successful(1))
+            self.assertFalse(is_executing_successful(3))
+
+    def test_sc_server(self):
+        class TestAgent(ScAgent):
+            def on_event(self, init_element: ScAddr, init_edge: ScAddr, action_element: ScAddr) -> ScResult:
+                finish_action_with_status(action_element, True)
+                return ScResult.OK
+
+        name = "some_agent"
+
+        def is_executing_successful() -> bool:
+            return execute_agent(
+                arguments={},
+                concepts=[],
+                initiation=name,
+                wait_time=WAIT_TIME,
+            )[1]
+
+        module = ScModule(TestAgent(name, ScEventType.ADD_OUTGOING_EDGE))
+        server.add_modules(module)
+        self.assertFalse(is_executing_successful())
+        with server.register_modules():
+            self.assertTrue(is_executing_successful())
+        self.assertFalse(is_executing_successful())
+
+        server.register_modules()
+        self.assertTrue(is_executing_successful())
+        server.remove_modules(module)
+        self.assertFalse(is_executing_successful())
+        server.add_modules(module)
+        server.unregister_modules()
+        self.assertFalse(is_executing_successful())
```

Comment:
can we add tests with delayed called agent?

Reply:
Done